lib/tests/detectors/virtualbox_detector.cc

```diff
@@ -0,0 +1,30 @@
+#include <catch.hpp>
+#include <internal/sources/dmi_source.hpp>
+#include <internal/detectors/virtualbox_detector.hpp>
+#include "../fixtures/dmi_fixtures.hpp"
+
+using namespace std;
+using namespace whereami::detectors;
+using namespace whereami::sources;
+using namespace whereami::testing::dmi;
+
+SCENARIO("Using the VirtualBox detector") {
+    WHEN("running in a VirtualBox VM") {
+        dmi_fixture_values dmi_virtualbox({
+            "VirtualBox",
+            "VirtualBox",
+            "Oracle Corporation",
+            "innotek GmbH",
+            "VirtualBox" });
+        THEN("it should return true") {
+            REQUIRE(virtualbox(dmi_virtualbox));
+        }
+    }
+
+    WHEN("running elsewhere") {
```

Comment:
it would be cool to add some fixtures for other hypervisors to this test case, but I think we can hold off on that until we add the detection for them.